.github/workflows/main-build.yml

```diff
@@ -61,6 +61,14 @@ jobs:
           java-version: 17
           distribution: temurin
 
+      - name: Cache local Maven repository
```

Comment:
this needs to go after we publish everything to maven local no? in line 72

Reply:
You insert the action at the point in which you want to restore the cache. GitHub actions will automatically cache the path after job success.